### ISSUE TITLE
Builder fails to build cloudflare module

### ISSUE THREAD

**Issue by sneycampos** [caddyserver/xcaddy#177]
I'm trying to build using xcaddy and cloudflare module but getting the following error:

`
44.94 /go/pkg/mod/github.com/caddyserver/caddy/v2@v2.7.5/modules/caddytls/automation.go:254:18: cannot use func(name string) error {…} (value of type func(name string) error) as func(ctx context.Context, name string) error value in struct literal
`

Dockerfile
```Dockerfile
FROM caddy:2.7.5-builder AS builder

RUN xcaddy build \
    --with github.com/caddy-dns/cloudflare

FROM caddy:2.7.5 AS runner

COPY --from=builder /usr/bin/caddy /usr/bin/caddy
```

Same problem was [reported in caddy-dns ](https://github.com/caddy-dns/cloudflare/issues/74)repository but i don't this is a error in cloudflare-dns module

I tried with alpine and non alpine images

**Comment by mohammed90** [comment]
This is not actually a bug in the builder. It's due to a breaking change in caddyserver/certmagic v0.20.0 library that Caddy had to accommodate. To build v2.7.5, use this command to force the older version of certmagic:

```
xcaddy build v2.7.5 --with github.com/caddyserver/certmagic@v0.19.2
```

**Comment by sneycampos** [comment]
> This is not actually a bug in the builder. It's due to a breaking change in caddyserver/certmagic v0.20.0 library that Caddy had to accommodate. To build v2.7.5, use this command to force the older version of certmagic:
> 
> ```
> xcaddy build v2.7.5 --with github.com/caddyserver/certmagic@v0.19.2
> ```
> 
> Any reason for sticking to v2.7.5 while v2.7.6 is already out?

i tried with 2.7.6 and got the same error, then tried with a minor version and cloudflare module said to use at least 2.7.5 so... 

thanks i will try with your solution and give you a feedback asap

**Comment by sneycampos** [comment]
> This is not actually a bug in the builder. It's due to a breaking change in caddyserver/certmagic v0.20.0 library that Caddy had to accommodate. To build v2.7.5, use this command to force the older version of certmagic:
> 
> ```
> xcaddy build v2.7.5 --with github.com/caddyserver/certmagic@v0.19.2
> ```
> 
> Any reason for sticking to v2.7.5 while v2.7.6 is already out?

FYI: in caddy builder 2.7.6 requires certmagic 0.20.0
`3.287 go: github.com/caddyserver/caddy/v2@v2.7.6 requires github.com/caddyserver/certmagic@v0.20.0, not github.com/caddyserver/certmagic@v0.19.2`

And thanks, it worked! :)

**Comment by mohammed90** [comment]
> > This is not actually a bug in the builder. It's due to a breaking change in caddyserver/certmagic v0.20.0 library that Caddy had to accommodate. To build v2.7.5, use this command to force the older version of certmagic:
> > ```
> > xcaddy build v2.7.5 --with github.com/caddyserver/certmagic@v0.19.2
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Any reason for sticking to v2.7.5 while v2.7.6 is already out?
> 
> FYI: in caddy builder 2.7.6 requires certmagic 0.20.0 `3.287 go: github.com/caddyserver/caddy/v2@v2.7.6 requires github.com/caddyserver/certmagic@v0.20.0, not github.com/caddyserver/certmagic@v0.19.2`
> 
> And thanks, it worked! :)

Yeah, so there was an issue with the cloudflare module, which my earlier fix (caddy-dns/cloudflare/#71) missed, i.e. downgrading certmagic as well.
